Patch-Source: https://github.com/facebook/watchman/commit/28d73f982a57159a1a60ad156720e1b0a8c73acd
--
From 28d73f982a57159a1a60ad156720e1b0a8c73acd Mon Sep 17 00:00:00 2001
From: Piotr Kubaj <pkubaj@FreeBSD.org>
Date: Tue, 6 Aug 2024 16:18:27 -0700
Subject: [PATCH] Fix build on big-endian architectures (#1238)

Summary:
Per https://docs.rs/bytes/1.6.1/bytes/buf/trait.BufMut.html#method.put_i16, there are no *_be functions, the ones for big-endian do not have any suffix.

Pull Request resolved: https://github.com/facebook/watchman/pull/1238

Reviewed By: chadaustin

Differential Revision: D60860963

Pulled By: kavehahmadi60

fbshipit-source-id: dff4fdc0aabb1b02a9ed96e601880c4941cf607d
---
 watchman/rust/serde_bser/src/ser/mod.rs | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/watchman/rust/serde_bser/src/ser/mod.rs b/watchman/rust/serde_bser/src/ser/mod.rs
index ceb85de618a1..58a805ba3e01 100644
--- a/vendor/serde_bser/src/ser/mod.rs
+++ b/vendor/serde_bser/src/ser/mod.rs
@@ -133,7 +133,7 @@ where
         #[cfg(target_endian = "little")]
         self.scratch.put_i16_le(v);
         #[cfg(target_endian = "big")]
-        self.scratch.put_i16_be(v);
+        self.scratch.put_i16(v);
     }
 
     #[inline]
@@ -143,7 +143,7 @@ where
         #[cfg(target_endian = "little")]
         self.scratch.put_i32_le(v);
         #[cfg(target_endian = "big")]
-        self.scratch.put_i32_be(v);
+        self.scratch.put_i32(v);
     }
 
     #[inline]
@@ -153,7 +153,7 @@ where
         #[cfg(target_endian = "little")]
         self.scratch.put_i64_le(v);
         #[cfg(target_endian = "big")]
-        self.scratch.put_i64_be(v);
+        self.scratch.put_i64(v);
     }
 }
 
@@ -255,7 +255,7 @@ where
         #[cfg(target_endian = "little")]
         self.scratch.put_f64_le(v);
         #[cfg(target_endian = "big")]
-        self.scratch.put_f64_be(v);
+        self.scratch.put_f64(v);
         Ok(())
     }
 
